dev-lang/ghc/ghc-9.2.7-r1.ebuild

```diff
@@ -30,29 +30,37 @@ BUILD_BIN_PV=9.0.2
 BUILD_BIN_REV=r4
 BUILD_BIN_PVR="${BUILD_BIN_PV}${BUILD_BIN_REV:+-${BUILD_BIN_REV}}"
 
+DEFAULT_SRC_HOST="https://eidetic.codes"
+
 ghc_binaries() {
-	local host="https://eidetic.codes"
 	echo "
-		binary? ( ${host}/${PN}-bin-${BIN_PVR}-${1}.gpkg.tar )
+		binary? ( ${DEFAULT_SRC_HOST}/${PN}-bin-${BIN_PVR}-${1}.gpkg.tar )
+		!binary? ( ${DEFAULT_SRC_HOST}/${PN}-bin-${BUILD_BIN_PVR}-${1}.gpkg.tar )
+	"
+}
+
+# Some arches have ghc-9.0 binaries packaged as .tar.gz, which doesn't fit the
+# pattern established in ghc_binaries()
+ghc_binaries_tar_build_bin() {
```

Comment:
Hm, maybe this should be renamed to `ghc_binaries_gz_build_bin()` to make it less ambiguous 😅